src/components/connector-authentication/useConnectorAuthentication.ts

```diff
@@ -1,81 +1,101 @@
-    const [authenticationResolvers, setAuthenticationResolvers] = useState<Omit<
-        // eslint-disable-next-line no-undef
-        PromiseWithResolvers<RefreshedAuthCredendentials | null>,
-        'promise'
-    > | null>(null);
-    const [executor, setExecutor] = useState<Executor | null>(null);
-    const [connectorName, setConnectorName] = useState<string>('');
-    const [result, setResult] = useState<ConnectorAuthenticationResult | null>(null);
+export type ConnectorPendingAuthentication = {
+    connectorId: string;
+    connectorName: string;
+    authenticationResolvers: Omit<PromiseWithResolvers<RefreshedAuthCredendentials | null>, 'promise'> | null;
+    executor: Executor | null;
+};
 
-    const resetProcess = useCallback(() => {
-        setAuthenticationResolvers(null);
-        setExecutor(null);
+export const useConnectorAuthentication = () => {
+    const [pendingAuthentications, setPendingAuthentications] = useState<ConnectorPendingAuthentication[]>([]);
+    const [authResults, setAuthResults] = useState<ConnectorAuthResult[]>([]);
+    const resetProcess = useCallback((id: string) => {
+        setPendingAuthentications((prev) => prev.filter((item) => item.connectorId !== id));
     }, []);
 
-    const process = useMemo(() => {
-        if (authenticationResolvers && executor) {
-            return {
-                __resolvers: authenticationResolvers,
-                async start() {
-                    setResult(null);
-                    try {
-                        const executorResult = await executor.handler().then((res) => {
-                            setResult(res);
-                            if (res.type === 'authentified') {
-                                return new RefreshedAuthCredendentials();
-                            }
-                            // eslint-disable-next-line no-console
-                            console.warn(`There is a "${res.type}" issue with authentifying of the connector`);
-                            if (res.type === 'error') {
+    const process = useCallback(
+        (id: string) => {
+            const authenticationProcess = pendingAuthentications.find((item) => item.connectorId === id);
+            if (!authenticationProcess) return null;
+
+            if (authenticationProcess.authenticationResolvers && authenticationProcess.executor) {
+                return {
+                    __resolvers: authenticationProcess.authenticationResolvers,
+                    async start() {
+                        try {
+                            const executorResult = await authenticationProcess.executor?.handler().then((res) => {
+                                setAuthResults((prev) => [
+                                    ...prev,
+                                    {
+                                        result: res,
+                                        connectorName: authenticationProcess.connectorName,
```

Comment:
I know it's rare real case, but since `connectorName` doesn't suppose to be unique, can we do identification based on `connectorId` instead?